tests/ctst/features/quotas/Quotas.feature

```diff
@@ -96,3 +96,43 @@ Feature: Quota Management for APIs
             | 100        | 200         | 0            | IAM_USER |
             | 100        | 0           | 200          | IAM_USER |
             | 100        | 200         | 200          | IAM_USER |
+
+    @2.6.0
+    @PreMerge
+    @Quotas
+    @CronJob
+    @DataDeletion
+    @NonVersioned
+    Scenario Outline: Quotas are affected by deletion operations between count items runs
+        Given an action "DeleteObject"
+        And a permission to perform the "PutObject" action
+        And a STORAGE_MANAGER type
+        And a bucket quota set to 10000 B
+        And an account quota set to 10000 B
+        And an upload size of 1000 B for the object "obj-1"
+        And a bucket quota set to <bucketQuota> B
+        And an account quota set to <accountQuota> B
+        And a <userType> type
+        And an environment setup for the API
+        And an "existing" IAM Policy that "applies" with "ALLOW" effect for the current API
+        When I wait 3 seconds
+        And I PUT an object with size <uploadSize>
+        Then the API should "fail" with "QuotaExceeded"
+        When the "count-items" cronjobs completes without error
+        When I wait 3 seconds
+        # At this point if negative inflights are not supported, write should
+        # not be possible, as the previous inflights are now part of the current
+        # metrics.
+        And i delete object "obj-1"
+        And I wait 3 seconds
+        And I PUT an object with size <uploadSize>
+        Then the API should "succeed" with ""
```

Comment:
we should also check that negative infligts do not allow to bypass the quota (i.e. we enforce `currnetUsage = max(currentReportedUsage + infligts, 0)`), i.e. if we had an pbject of size 1000, set of quota to 200, then delete the object : we must not be able to push another object of size 1000...

Reply:
I added a scenario, the same, but at the end, instead of putting 100B (where we have 1000 current and -1000 inflights, which is thus under the 200B quota, 1000-1000+100 < 200), we put 1000B, and expect a quota exceeded error.

Reply:
that scenario would only reach 0, not a negative value though...

what I am worried is kind of race conditions where we could have `countItemsMetrics - inflight < 0` for a few instants : i.e. `0 - 1000 + x < 200` would allow up to x-1200... while for additional safety we should have in the code `max(countItemsValue + inflightValue, 0) < quota`. This was not relevant when inflights were >=0, but we should have that extra check now (preventively)

Reply:
I am not sure to understand how we can reach such a state: event if the inflights are not applied directly, s3 ensures we cannot delete more objects that what we have, so if we have 0 "current" and -1000 inflights, it means we accepted to delete 1000 bytes from an object that was not part of the metrics, which is not possible?
At the very least, that is not something we can test here.

Reply:
At the end, rejecting any negative global current+inflight may block the client in case we really missed some (existing) s3 object in the "current", and we would have to wait for the next count items run. Unless we suspect we can delete more data than what we really have in s3, which is confusing

Reply:
For
```
max(countItemsValue + inflightValue, 0) < quota
```

I am not sure it helps? If we have 0 current, -1000 inflight and try to write:
- `0` -> allowed
- Any value between 1 and 1200 -> allowed (we get a value <= quota). E.g. `Math.max(0 -1000 + 1200, 0) = 200 < 200`
- Any value above -> denied

At the end the check does not add any more protection than what we already have, that is, without the `Math.max`?

Reply:
If we only want to accept the "200" after the 0, then we are doing the same as before: we don't accept negative inflights. If we just want to accept the 0-1000 range, then the formula should be updated, but to me it does not make sense...

Reply:
it is not about rejecting negative infligts, but about ensuring we never think the "current" usage (i.e. last reported countItem size + inflights) should never be < 0.... which could allow accepting more than the quota!

Reply:
As discussed I am closing this for now as we won't do it today